<a href="https://colab.research.google.com/github/xkaycc/SALES-PREDICTION/blob/main/Project_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## **Project 2**

### **Task**
- propose two possible datasets
- choose your first choice data set, and a backup data set 

**First and second choice: dataset 1&2**

1. Source of data

2. Brief description of data

3. What is the target?

4. What does one row represent? (A person?  A business?  An event? A product?)

5. Is this a classification or regression problem?

6. How many features does the data have?

7. How many rows are in the dataset?

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?

## **Dataset 1**

**This dataset is titled medical cost insurance data and I will be using it for regression analysis** 

In [ ]:
##Import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [ ]:
#load dataset 1 (insurance data)
df_1 = pd.read_csv(r"C:\Users\USER\Desktop\crude oil.csv")

In [ ]:
#check first 5 rows of df_1
df_1.head()

ID                 time  generation biomass  \
0  ID_000000  2019-01-01 00:00:00               447.0   
1  ID_000001  2019-01-01 01:00:00               449.0   
2  ID_000002  2019-01-01 02:00:00               448.0   
3  ID_000003  2019-01-01 03:00:00               438.0   
4  ID_000004  2019-01-01 04:00:00               428.0   

   generation fossil brown coal/lignite  generation fossil coal-derived gas  \
0                                 329.0                                 0.0   
1                                 328.0                                 0.0   
2                                 323.0                                 0.0   
3                                 254.0                                 0.0   
4                                 187.0                                 0.0   

   generation fossil gas  generation fossil hard coal  generation fossil oil  \
0                 4844.0                       4821.0                  162.0   
1                 5196.0                       4755.0                  158.0   
2                 4857.0                       4581.0                  157.0   
3                 4314.0                       4131.0                  160.0   
4                 4130.0                       3840.0                  156.0   

   generation fossil oil shale  generation fossil peat  ...  generation waste  \
0                          0.0                     0.0  ...             196.0   
1                          0.0                     0.0  ...             195.0   
2                          0.0                     0.0  ...             196.0   
3                          0.0                     0.0  ...             191.0   
4                          0.0                     0.0  ...             189.0   

   generation wind offshore  generation wind onshore  \
0                       0.0                   6378.0   
1                       0.0                   5890.0   
2                       0.0                   5461.0   
3                       0.0                   5238.0   
4                       0.0                   4935.0   

   forecast solar day ahead  forecast wind offshore eday ahead  \
0                      17.0                                NaN   
1                      16.0                                NaN   
2                       8.0                                NaN   
3                       2.0                                NaN   
4                       9.0                                NaN   

   forecast wind onshore day ahead  total load forecast  total load actual  \
0                           6436.0              26118.0            25385.0   
1                           5856.0              24934.0            24382.0   
2                           5454.0              23515.0            22734.0   
3                           5151.0              22642.0            21286.0   
4                           4861.0              21785.0            20264.0   

   price day ahead  price actual  
0           5010.0        6541.0  
1           4810.0        6492.0  
2           4733.0        6448.0  
3           4227.0        5932.0  
4           3841.0        5604.0  

[5 rows x 30 columns]

In [ ]:
#check the shape of df_1
df_1.shape
print (f' There are {df_1.shape[0]} rows and {df_1.shape[1]} columns')

 There are 27500 rows and 30 columns


## **Source of the Dataset**

**This dataset was gotten from Zindi Africa site**

## **Brief description of data**

**This dataset is having the data of 27K+ crude oil and can we accurately predict price actual?**

- Dataset Walkthrough
- Understanding Dataset Hierarchy
- Data Preprocessing
- Exploratory Data Analysis
- Data Visualization
- Making Recommendation System

## **What is the target?**

- price actual

## **What does one row represent?**

- Each row represent a product

## **Is this a classification or regression problem?**

- Regression

## **How many features does the data have?**

- There are 29 features aside the target variable

## **How many rows are in the dataset?**

- 27500 rows

## **What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?**

In [ ]:
## Check for missing values
df_1.isnull().sum()

ID                                                 0
time                                               0
generation biomass                                18
generation fossil brown coal/lignite              17
generation fossil coal-derived gas                17
generation fossil gas                             17
generation fossil hard coal                       17
generation fossil oil                             18
generation fossil oil shale                       17
generation fossil peat                            17
generation geothermal                             17
generation hydro pumped storage aggregated     27500
generation hydro pumped storage consumption       18
generation hydro run-of-river and poundage        18
generation hydro water reservoir                  17
generation marine                                 18
generation nuclear                                16
generation other                                  17
generation other renewable                    

- There alot of missing data

In [ ]:
## Have a look at the descriptive statistics
df_1.describe()

generation biomass  generation fossil brown coal/lignite  \
count         27482.00000                          27483.000000   
mean            396.78022                            476.247098   
std              88.97453                            363.187665   
min               0.00000                              0.000000   
25%             340.00000                              0.000000   
50%             376.00000                            558.000000   
75%             475.00000                            856.000000   
max             592.00000                            999.000000   

       generation fossil coal-derived gas  generation fossil gas  \
count                             27483.0           27483.000000   
mean                                  0.0            5555.281119   
std                                   0.0            2252.850015   
min                                   0.0            1518.000000   
25%                                   0.0            4069.000000   
50%                                   0.0            4871.000000   
75%                                   0.0            6269.000000   
max                                   0.0           20034.000000   

       generation fossil hard coal  generation fossil oil  \
count                 27483.000000           27482.000000   
mean                   4411.479242             303.007059   
std                    2015.548483              53.850784   
min                       0.000000               0.000000   
25%                    2618.000000             267.000000   
50%                    4698.000000             303.000000   
75%                    6062.500000             336.000000   
max                    8359.000000             449.000000   

       generation fossil oil shale  generation fossil peat  \
count                      27483.0                 27483.0   
mean                           0.0                     0.0   
std                            0.0                     0.0   
min                            0.0                     0.0   
25%                            0.0                     0.0   
50%                            0.0                     0.0   
75%                            0.0                     0.0   
max                            0.0                     0.0   

       generation geothermal  generation hydro pumped storage aggregated  ...  \
count                27483.0                                         0.0  ...   
mean                     0.0                                         NaN  ...   
std                      0.0                                         NaN  ...   
min                      0.0                                         NaN  ...   
25%                      0.0                                         NaN  ...   
50%                      0.0                                         NaN  ...   
75%                      0.0                                         NaN  ...   
max                      0.0                                         NaN  ...   

       generation waste  generation wind offshore  generation wind onshore  \
count      27482.000000                   27483.0             27483.000000   
mean         261.802998                       0.0              5469.452061   
std           51.346693                       0.0              3194.413273   
min            0.000000                       0.0                 0.000000   
25%          228.000000                       0.0              2968.500000   
50%          269.000000                       0.0              4851.000000   
75%          303.000000                       0.0              7367.000000   
max          357.000000                       0.0             17436.000000   

       forecast solar day ahead  forecast wind offshore eday ahead  \
count              27500.000000                                0.0   
mean                1439.292509                                NaN   
std                 1677.600341       

## **Summary**

- From the little analysis, I believe this will be tasking work that will require alot of cleaning. 
- Modeling too will be tasking to the best model that best fit

# **Dataset 2**

**This dataset is titled hotel reservations data and I will be using it for classification analysis**

In [ ]:
#load dataset 1 (insurance data)
df_2 = pd.read_csv(r"C:\Users\USER\Desktop\Hotel Reservations.csv")

In [ ]:
#check first 5 rows of df_1
df_2.head()

Booking_ID  no_of_adults  no_of_children  no_of_weekend_nights  \
0   INN00001             2               0                     1   
1   INN00002             2               0                     2   
2   INN00003             1               0                     2   
3   INN00004             2               0                     0   
4   INN00005             2               0                     1   

   no_of_week_nights type_of_meal_plan  required_car_parking_space  \
0                  2       Meal Plan 1                           0   
1                  3      Not Selected                           0   
2                  1       Meal Plan 1                           0   
3                  2       Meal Plan 1                           0   
4                  1      Not Selected                           0   

  room_type_reserved  lead_time  arrival_year  arrival_month  arrival_date  \
0        Room_Type 1        224          2017             10             2   
1        Room_Type 1          5          2018             11             6   
2        Room_Type 1          1          2018              2            28   
3        Room_Type 1        211          2018              5            20   
4        Room_Type 1         48          2018              4            11   

  market_segment_type  repeated_guest  no_of_previous_cancellations  \
0             Offline               0                             0   
1              Online               0                             0   
2              Online               0                             0   
3              Online               0                             0   
4              Online               0                             0   

   no_of_previous_bookings_not_canceled  avg_price_per_room  \
0                                     0               65.00   
1                                     0              106.68   
2                                     0               60.00   
3                                     0              100.00   
4                                     0               94.50   

   no_of_special_requests booking_status  
0                       0   Not_Canceled  
1                       1   Not_Canceled  
2                       0       Canceled  
3                       0       Canceled  
4                       0       Canceled

In [ ]:
#check the shape of df_1
df_2.shape
print (f' There are {df_2.shape[0]} rows and {df_2.shape[1]} columns')

 There are 36275 rows and 19 columns


## **Source of the Dataset**

This dataset was gotten from kaggle site https://www.kaggle.com/datasets/ahsan81/hotel-reservations-classification-dataset

## **Brief description of data**

The online hotel reservation channels have dramatically changed booking possibilities and customers’ behavior. A significant number of hotel reservations are called-off due to cancellations or no-shows. The typical reasons for cancellations include change of plans, scheduling conflicts, etc. This is often made easier by the option to do so free of charge or preferably at a low cost which is beneficial to hotel guests but it is a less desirable and possibly revenue-diminishing factor for hotels to deal with.

- **Can you predict if the customer is going to honor the reservation or cancel it?**

## **What is the target?**

- Booking status

## **What does one row represent?**

- Business

## **Is this a classification or regression problem?**

- Classification

## **How many features does the data have?**

- There are eighteen(18) features aside the target variable

## **How many rows are in the dataset?**

- 36275 rows

## **What, if any, challenges do you foresee in cleaning, exploring, or modeling with this dataset?**

In [ ]:
## Check for missing values
df_2.isnull().sum()

Booking_ID                              0
no_of_adults                            0
no_of_children                          0
no_of_weekend_nights                    0
no_of_week_nights                       0
type_of_meal_plan                       0
required_car_parking_space              0
room_type_reserved                      0
lead_time                               0
arrival_year                            0
arrival_month                           0
arrival_date                            0
market_segment_type                     0
repeated_guest                          0
no_of_previous_cancellations            0
no_of_previous_bookings_not_canceled    0
avg_price_per_room                      0
no_of_special_requests                  0
booking_status                          0
dtype: int64

**No missing values**

In [ ]:
## Have a look at the descriptive statistics
df_2.describe()

no_of_adults  no_of_children  no_of_weekend_nights  no_of_week_nights  \
count  36275.000000    36275.000000          36275.000000       36275.000000   
mean       1.844962        0.105279              0.810724           2.204300   
std        0.518715        0.402648              0.870644           1.410905   
min        0.000000        0.000000              0.000000           0.000000   
25%        2.000000        0.000000              0.000000           1.000000   
50%        2.000000        0.000000              1.000000           2.000000   
75%        2.000000        0.000000              2.000000           3.000000   
max        4.000000       10.000000              7.000000          17.000000   

       required_car_parking_space     lead_time  arrival_year  arrival_month  \
count                36275.000000  36275.000000  36275.000000   36275.000000   
mean                     0.030986     85.232557   2017.820427       7.423653   
std                      0.173281     85.930817      0.383836       3.069894   
min                      0.000000      0.000000   2017.000000       1.000000   
25%                      0.000000     17.000000   2018.000000       5.000000   
50%                      0.000000     57.000000   2018.000000       8.000000   
75%                      0.000000    126.000000   2018.000000      10.000000   
max                      1.000000    443.000000   2018.000000      12.000000   

       arrival_date  repeated_guest  no_of_previous_cancellations  \
count  36275.000000    36275.000000                  36275.000000   
mean      15.596995        0.025637                      0.023349   
std        8.740447        0.158053                      0.368331   
min        1.000000        0.000000                      0.000000   
25%        8.000000        0.000000                      0.000000   
50%       16.000000        0.000000                      0.000000   
75%       23.000000        0.000000                      0.000000   
max       31.000000        1.000000                     13.000000   

       no_of_previous_bookings_not_canceled  avg_price_per_room  \
count                          36275.000000        36275.000000   
mean                               0.153411          103.423539   
std                                1.754171           35.089424   
min                                0.000000            0.000000   
25%                                0.000000           80.300000   
50%                                0.000000           99.450000   
75%                                0.000000          120.000000   
max                               58.000000          540.000000   

       no_of_special_requests  
count            36275.000000  
mean                 0.619655  
std                  0.786236  
min                  0.000000  
25%                  0.000000  
50%                  0.000000  
75%                  1.000000  
max                  5.000000

In [ ]:
df_2.select_dtypes(exclude = np.number).columns

Index(['Booking_ID', 'type_of_meal_plan', 'room_type_reserved',
       'market_segment_type', 'booking_status'],
      dtype='object')

In [ ]:
df_2.describe(exclude = np.number)

Booking_ID type_of_meal_plan room_type_reserved market_segment_type  \
count       36275             36275              36275               36275   
unique      36275                 4                  7                   5   
top      INN20042       Meal Plan 1        Room_Type 1              Online   
freq            1             27835              28130               23214   

       booking_status  
count           36275  
unique              2  
top      Not_Canceled  
freq            24390

## **Summary 2**

- From the little analysis, there will be alot of modifications and challenges during the course of the work. 
- No much cleaning but alot of exploration is needed
- considering it is large dataset with many columns, some columns might make the model to underfit or overfit.